cmd/snap/cmd_set.go

```diff
@@ -155,6 +161,34 @@ func (x *cmdSet) Execute([]string) error {
 	return nil
 }
 
+func validateJSONKeys(v interface{}) error {
+	stack := []interface{}{v}
+
+	for len(stack) > 0 {
+		current := stack[len(stack)-1]
+		stack = stack[:len(stack)-1]
+
+		switch current := current.(type) {
+		case map[string]interface{}:
+			for key, value := range current {
+				if err := config.ValidateKey(key); err != nil {
+					return err
+				}
+
+				stack = append(stack, value)
+			}
+
+		case []interface{}:
+			stack = append(stack, current...)
+		case []map[string]interface{}:
+			for _, item := range current {
+				stack = append(stack, item)
+			}
```

Comment:
So, after a bit of testing, the conclusion is that `[]interface{}` and `[]map[string]interface{}` are distinct types, and there isn't any type that can potentially take both branches, since the slices must have different memory layouts. So, for correctness, I think both cases should be there. However, I do not think that the `json` package can ever give us back a `[]map[string]interface{}`, as long as we are unmarshalling into an `interface{}`.

Reply:
My vote would then be this:
```suggestion
		case []interface{}, []map[string]interface:
			// json package always produces []interface{} for any list,
			// but match []map[string]interface{} as well just in case
			stack = append(stack, current...)
```

I still think if we have `[]map[string]interface`, then we should also have `[][]interface{}`, and probably also `[][]map[string]interface{}`, etc. But obviously that's untenable. So this is reasonable enough, especially with a comment.

Reply:
If you use like this: `[]map[string]interface` you got a syntax error problem: `expected '{', found ':'`

If you correct the previous error putting the `{}` like `[]map[string]interface{}`, there it's not possible anymore to use the variadic parameter on `current`.